mslib/utils/verify_waypoint_data.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""
+
+    mslib.utils.verify_waypoint_data
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    basic checks for xml waypoint data.
+
+    This file is part of MSS.
+
+    :copyright: Copyright 2024 Reimar Bauer
+    :license: APACHE-2.0, see LICENSE for details.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+
+import xml.dom.minidom
```

Comment:
This might be a more fundamental issue: the xml processing modules from python's stdlib are _not_ suitable to parse untrusted xml documents. See the first warning here: https://docs.python.org/3/library/xml.html.

My understanding is that we allow users to upload those xml documents directly, so a malicious user might be able to craft the xml in such a way that it e.g. crashes the MSColab server.